lib/editor/actions/map/stopStrategies.js

```diff
@@ -550,14 +552,16 @@ export function removeStopFromPattern (pattern: Pattern, stop: GtfsStop, index:
         dispatch(setErrorMessage({message: `Could not remove stop from pattern:`}))
         return
       }
+      patternStops.splice(index, 1)
```

Comment:
Why is this extracted from being run after the `if/else` block to running both inside the if and else block?

Reply:
I had extracted it so that the `updateShapeDistTraveled` method would have the "correct" set of pattern stops when called. However, that method also looks up the stop using a filter on stopid, so splicing the stop out of there before hand doesn't actually matter. From a code perspective we could move the splice back after, but it may be more confusing inside of the `updateShapeDistTraveled` method since the patternStops will be "out of step" with the segments and controlPoints. I think as is might be easier to maintain.